File: ReverseArray2.py
def reverseArray(arr):
    i = 0
    j = len(arr) - 1
    while(i < j):
        temp = arr[i]
        arr[i] = arr[j]
        arr[j] = temp
        i = i + 1
        j = j - 1
    return arr

my_list = [3,5,4,3,6,5,4]
output = reverseArray(my_list)
print(output)

File: ReverseArray2.java
public class ReverseArray2 {
    static void swapInArray(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    static void reverseArray(int arr[]) {
        int i = 0;
        int j = arr.length - 1;
        while (i < j) {
            swapInArray(arr, i, j);
            i++;
            j--;
        }
    }

    static void printArray(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }

    }

    public static void main(String[] args) {
        int[] arr = { 5, 8, 3, 4, 7 };
        reverseArray(arr);
        printArray(arr);
    }

}
